packages/core/src/scripts/migrate-oz-sdk-project.ts

```diff
@@ -0,0 +1,351 @@
+  const impls: Record<string, ImplDeployment> = {};
+  for (const contractName in contracts) {
+    const contract = contracts[contractName];
+    if (contract.deployedBytecodeHash === undefined) {
+      continue;
+    } else {
+      impls[contract.deployedBytecodeHash] = transformImplementationItem(contract);
+    }
+  }
+  return impls;
+}
+
+function transformImplementationItem(contract: LegacyContract): ImplDeployment {
+  if (contract.address === undefined) {
+    throw new Error('Could not find implementation address');
+  }
+
+  return {
+    address: contract.address,
+    layout: transformLayout(contract),
+  };
+}
+
+function transformLayout(contract: LegacyContract): StorageLayout {
+  return {
+    storage: contract.storage?.map(transformStorageItem) || [],
+    types: transformTypes(contract.types || {}),
+  };
+}
+
+function transformStorageItem(storageItem: LegacyStorageItem): StorageItem {
+  return {
+    contract: storageItem.contract,
+    label: storageItem.label,
+    type: transformTypeName(storageItem.type),
+    // TODO reconstruct path and line if sourcecode is available
+    src: storageItem.path,
+  };
+}
+
+function transformTypes(oldTypes: LegacyTypes): Record<string, TypeItem> {
+  const newTypes: Record<string, TypeItem> = {};
+  for (const typeName in oldTypes) {
+    newTypes[transformTypeName(typeName)] = transformType(getTypeKind(typeName), oldTypes[typeName]);
+  }
+  return newTypes;
+}
+
+function transformType(typeKind: string, oldType: LegacyType): TypeItem {
+  switch (typeKind) {
+    case 'Struct':
+      return {
+        label: stripContractName(oldType.label),
+        members: (oldType.members as StructMember[]).map(member => ({
+          label: stripContractName(member.label),
+          type: transformTypeName(member.type),
+        })),
+      };
+    case 'Enum':
+      return {
+        label: stripContractName(oldType.label),
+        members: oldType.members,
+      };
+    default:
+      return {
+        label: stripContractName(oldType.label),
+      };
+  }
+}
+
+function transformTypeName(typeName: string): string {
+  switch (getTypeKind(typeName)) {
+    case 'Struct':
+      return transformStructTypeName(typeName);
+    case 'Enum':
+      return transformEnumTypeName(typeName);
+    case 'Mapping':
+      return transformMappingTypeName(typeName);
+    case 'DynArray':
+      return transformDynArrayTypeName(typeName);
+    case 'StaticArray':
+      return transformStaticArrayTypeName(typeName);
+    case 'Elementary':
+      return typeName;
+    default:
+      throw new Error(`Unknown type: ${typeName}`);
+  }
+}
+
+function transformStructTypeName(typeName: string): string {
+  const valueType = stripContractName(getValueType(typeName));
+  return `t_struct(${valueType})_storage`;
+}
+
+function transformEnumTypeName(typeName: string): string {
+  const valueType = stripContractName(getValueType(typeName));
+  return `t_enum(${valueType})`;
+}
```

Comment:
Enums and structs also have the AST id as part of their name in the plugins. For example `t_enum(Direction)92` and `t_struct(Foo)89_storage`.

I thought we had this information but I think I was mistaken. The SDK stores the AST ids for the state variables, but not for the struct definition.

In any case, it's not critical, everything should work without the id. We should respect the same format, though, so we have to make up a number and put it in. This number we make up should be unique in the same way that AST ids would be unique, so that future code can rely on that. (By this I mean we can't just hardcode the number 42 everywhere.)

This should be doable but it may require some changes to the code because the made up number has to be propagated for each type... unless it can be generated deterministically from the name alone. What do you think?